tests/test_cobs.cc

```diff
@@ -1,213 +1,206 @@
 #include "cobs.h"
+#include <cassert>
+#include <vector>
 #include <cstring>
 
-/*
-Throw exceptions so we can catch them with gdb (catch catch).
+/**
+Fill a byte buffer with a given value.
 */
-void throw_assert(bool condition, const char * const remark) {
-    if (condition == false) {
-        throw remark;
-    }
+void fill(uint8_t * const buffer, size_t count, uint8_t value) {
+    std::memset(buffer, value, count);
 }
 
-/*
-Fill a byte buffer with a given value.
+/**
+A structure that represents a sequence of bytes.
 */
-void fill(uint8_t * const buffer, size_t length, uint8_t value) {
-    std::memset(buffer, value, length);
-}
+struct Rep {
+    size_t count;
+    uint8_t value;
+};
 
-/*
-Check that a byte buffer contains only a given value.
+/**
+Fill a byte buffer with given repetitions.
 */
-void assert_mem_is(uint8_t * buffer, size_t length, uint8_t value) {
-    uint8_t * buffer_end = buffer + length;
-    while (buffer < buffer_end) {
-        throw_assert(*(buffer++) == value, "");
+size_t fill_repetitions(uint8_t * const buffer, std::vector<Rep> repetitions) {
+    size_t offset = 0;
+
+    for (auto const& rep: repetitions) {
+        fill(buffer + offset, rep.count, rep.value);
```

Comment:
you can resize instead: http://en.cppreference.com/w/cpp/container/vector/resize